packages/decentraland-ecs/types/@decentraland/Identity/index.d.ts

```diff
@@ -1,9 +1,29 @@
 declare module '@decentraland/Identity' {
+  export type ColorString = string
+
+  export type Snapshots = {
+    face: string
+    face256: string
+    face128: string
+    body: string
+  }
+
+  export type AvatarForUserData = {
+    bodyShape: WearableId
+    skinColor: ColorString
+    hairColor: ColorString
+    eyeColor: ColorString
+    wearables: WearableId[]
+    snapshots: Snapshots
+  }
+
   export type UserData = {
     displayName: string
-    publicKey: string
+    publicKey: string | null
```

Comment:
Was this always null?

Reply:
At least since july 2020 (https://github.com/decentraland/explorer/blob/6b83254c330cc284352d79166da1df17df392054/kernel/packages/shared/apis/UserIdentity.ts)
`publicKey` can be `null`